openstack/imageservice/v2/images/results.go

```diff
@@ -0,0 +1,404 @@
+		return asString(any, k)
+	}
+	return "", fmt.Errorf("expected key \"%s\" in map, but this key is not present", k)
+}
+
+func extractNoneableStringAtKey(m map[string]interface{}, k string) (string, error) {
+	if any, ok := m[k]; ok {
+		return asNoneableString(any, k)
+	}
+	return "", fmt.Errorf("expected key \"%s\" in map, but this key is not present", k)
+}
+
+func extractNoneableIntegerAtKey(m map[string]interface{}, k string) (int, error) {
+	if any, ok := m[k]; ok {
+		return asNoneableInteger(any, k)
+	}
+	return 0, fmt.Errorf("expected key \"%s\" in map, but this key is not present", k)
+}
+
+func extractStringSliceAtKey(m map[string]interface{}, key string) ([]string, error) {
+	if any, ok := m[key]; ok {
+		if slice, ok := any.([]interface{}); ok {
+			res := make([]string, len(slice))
+			for k, v := range slice {
+				var err error
+				if res[k], err = asString(v, key); err != nil {
+					return nil, err
+				}
+			}
+			return res, nil
+		}
+		return nil, fmt.Errorf("expected slice as \"%s\" value, but found: %#v", key, any)
+	}
+	return nil, fmt.Errorf("expected key \"%s\" in map, but this key is not present", key)
+}
+
+func stringToImageStatus(s string) (ImageStatus, error) {
+	if s == "queued" {
+		return ImageStatusQueued, nil
+	} else if s == "active" {
+		return ImageStatusActive, nil
+	} else {
+		return "", fmt.Errorf("expected \"queued\" or \"active\" as image status, but found: \"%s\"", s)
+	}
+}
+
+func extractImageStatusAtKey(m map[string]interface{}, k string) (ImageStatus, error) {
+	if any, ok := m[k]; ok {
+		if str, ok := any.(string); ok {
+			return stringToImageStatus(str)
+		}
+		return "", fmt.Errorf("expected string as \"%s\" value, but found: %#v", k, any)
+	}
+	return "", fmt.Errorf("expected key \"%s\" in map, but this key is not present", k)
+
+}
+
+func stringToImageVisibility(s string) (ImageVisibility, error) {
+	if s == "public" {
+		return ImageVisibilityPublic, nil
+	} else if s == "private" {
+		return ImageVisibilityPrivate, nil
+	} else {
+		return "", fmt.Errorf("expected \"public\" or \"private\" as image status, but found: \"%s\"", s)
+	}
+}
+
+func extractImageVisibilityAtKey(m map[string]interface{}, k string) (ImageVisibility, error) {
+	if any, ok := m[k]; ok {
+		if str, ok := any.(string); ok {
+			return stringToImageVisibility(str)
+		}
+		return "", fmt.Errorf("expected string as \"%s\" value, but found: %#v", k, any)
+	}
+	return "", fmt.Errorf("expected key \"%s\" in map, but this key is not present", k)
+}
+
+func extractBoolAtKeyOptional(m map[string]interface{}, k string, ifMissing bool) (bool, error) {
+	if any, ok := m[k]; ok {
+		return asBool(any)
+	}
+	return ifMissing, nil
+
+}
+
+func extractMapStringStringAtKeyOptional(m map[string]interface{}, k string, ifMissing map[string]string) (map[string]string, error) {
+	if any, ok := m[k]; ok {
+		return asMapStringString(any)
+	}
+	return ifMissing, nil
+}
+
+func extractImage(res gophercloud.Result) (*Image, error) {
```

Comment:
This is something I took from the original PR and to be honest I'm not sure why this is needed.
I pushed a simplification which passes unit tests and my functional testing as well.